rascal-textmate-core/src/main/rascal/lang/textmate/ConversionUnit.rsc

```diff
@@ -137,4 +139,95 @@ private list[tuple[Keygen, Compare]] sorters = [
 
     // Sort by stringified production
     <getStringifiedProduction, bool(str s1, str s2) { return s1 < s2; }>
-];
\ No newline at end of file
+];
+
+@synopsis{
+    Retains from set `units` each unit that is a prefix (i.e., the list of
+    symbols of its production) of any other unit in `units`
+}
+
+set[ConversionUnit] retainStrictPrefixes(set[ConversionUnit] units)
+    = {u1 | u1 <- units, any(u2 <- units, u1 != u2, isStrictPrefix(u1, u2))};
+
+@synopsis{
+    Removes from set `units` each units that is a prefix (i.e., the list of
+    symbols of its production) of any other unit in `units`
+}
+
+set[ConversionUnit] removeStrictPrefixes(set[ConversionUnit] units)
+    = units - retainStrictPrefixes(units);
+
+@synopsis{
+    Checks if unit `u1` is a strict prefix of unit `u2`
+}
+
+bool isStrictPrefix(ConversionUnit u1, ConversionUnit u2)
+    = isStrictPrefix(u1.prod.symbols, u2.prod.symbols);
+
+// TODO: This function could be moved to a separate, generic module
+private bool isStrictPrefix([], [])
+    = false;
+private bool isStrictPrefix([], [_, *_])
+    = true;
+private bool isStrictPrefix([_, *_], [])
+    = false;
+private bool isStrictPrefix([head1, *tail1], [head2, *tail2])
+    = head1 == head2 && isStrictPrefix(tail1, tail2);
```

Comment:
this is very functional way of saying it ;) I get it, but it took me a while. I would have done aloop over an index, but thats all up to the person writing it I guess. 

(I don't prefer the messy way of saying, hey, if the list 1 is smaller than list 2, they aren't a strict prefix).

```rascal
bool isStrictPrefix(list[&T] a, list[&T] b) {
  if (size(a) >= size(b) {
     return false;
  }
  for (i <- [0..size(a)], a[i] != b[i]) {
     return false;
  }
  return true;
}
```

would have been how I wrote it.

Reply:
Alright, two can play that game 😉 I went with:

```rascal
bool isStrictPrefix(list[&T] a, list[&T] b)
  = size(a) < size(b) && !any(i <- [0..size(a)], a[i] != b[i]);
```

Reply:
Even better 👍🏼